master/internal/task/tasklogger/logger_test.go

```diff
@@ -0,0 +1,152 @@
+package tasklogger_test
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/slices"
+
+	"github.com/determined-ai/determined/master/internal/task/tasklogger"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+type arrayWriter struct {
```

Comment:
also waffled here between using mockery and this, but the `mock.Call` field in mockery is not threadsafe so i can't check it to see if all the calls have been made while they are happening and i want to use `-race` to run the tests.